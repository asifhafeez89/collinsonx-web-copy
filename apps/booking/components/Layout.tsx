import { Box, Center, Container } from '@collinsonx/design-system/core';

import { ReactNode, Ref, forwardRef } from 'react';
import usePayload from 'hooks/payload';
import AppLogo from './AppLogo';
import colors from 'ui/colour-constants';

interface LayoutProps {
  children: ReactNode;
}
type ContainerRef = Ref<HTMLDivElement> | undefined;

const Layout = forwardRef(({ children }: LayoutProps, ref: ContainerRef) => {
  const { payload, setPayload } = usePayload();

  return (
    <Container
      px={0}
      sx={{
        maxWidth: '100%',
        height: '100%',

        '@media (max-width: 768px)': {
          margin: '0',
          padding: '0',
        },
        overflow: 'scroll',
      }}
      ref={ref}
    >
      <Box
        sx={{
          width: '100%',
          backgroundColor: colors.white,
          boxShadow: `4px 4px 4px 0px ${colors.shadow}`,
          position: 'fixed',
          zIndex: 200,
        }}
      >
        <Box
          mb={2}
          mt={2}
          sx={{
            width: '100%',
            backgroundColor: colors.white,
            boxShadow: `4px 4px 4px 4px ${colors.shadow}`,
            position: 'relative',
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            alignContent: 'center',
            alignItems: 'center',
            height: '100px',
            '@media (max-width: 768px)': {
              height: '50px',
            },
            zIndex: 200,
          }}
        >
          {payload && (
            <AppLogo
              accountProvider={payload.accountProvider}
              membershipType={payload.membershipType}
            />
          )}
        </Box>
      </Box>
      <Box
        sx={{
          paddingBottom: '1.3rem',
          marginTop: '6.4rem',

          '@media (max-width: 768px)': {
            marginTop: '3.rem',
            padding: '0 0 1.3rem 0',
          },
        }}
      >
        {children}
      </Box>
    </Container>
  );
});

export default Layout;
