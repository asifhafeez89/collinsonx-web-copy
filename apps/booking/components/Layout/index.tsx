import { Box, Center, Container } from '@collinsonx/design-system/core';

import { ReactNode, Ref, forwardRef } from 'react';
import usePayload from 'hooks/payload';
import AppLogo from '../AppLogo';

import classes from './Layout.module.css';

interface LayoutProps {
  children: ReactNode;
}
type ContainerRef = Ref<HTMLDivElement> | undefined;

const Layout = forwardRef(({ children }: LayoutProps, ref: ContainerRef) => {
  const { payload, setPayload } = usePayload();

  return (
    <Container px={0} fluid className={classes.container} ref={ref}>
      <Box className={classes.containerL2}>
        <Box mb={2} mt={2} className={classes.logoWrapper}>
          {payload && (
            <AppLogo
              accountProvider={payload.accountProvider}
              membershipType={payload.membershipType}
            />
          )}
        </Box>
      </Box>
      <Box className={classes.contentWrapper}>{children}</Box>
    </Container>
  );
});

export default Layout;
