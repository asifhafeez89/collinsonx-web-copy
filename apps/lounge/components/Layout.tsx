import { Container, MantineProvider } from '@collinsonx/design-system/core';
import { Header, experienceX } from '@collinsonx/design-system';
import { Be_Vietnam_Pro } from '@next/font/google';
import { LogoExperienceX } from '@collinsonx/design-system/assets/logo';

import { Cart, Chat, Home } from '@collinsonx/design-system/assets/icons';
import useAuth from '@collinsonx/utils/hooks/useAuth';

const beVietnamPro = Be_Vietnam_Pro({
  style: ['normal'],
  subsets: ['latin'],
  weight: ['400', '600', '700'],
});

interface LayoutProps {
  children: JSX.Element;
}

export default function Layout({ children }: LayoutProps) {
  const [isLoggedIn, userId, logout] = useAuth({});

  const handleLogout = async () => {
    if (typeof logout === 'function') {
      console.log('---- Logout await');
      await logout();
      console.log('---- Logout redirect');
      window.location.href = '/';
    }
  };
  return (
    <>
      <Header
        onClickSignout={handleLogout}
        logo={<LogoExperienceX />}
        items={[
          {
            label: 'Home',
            link: '/lounge',
            icon: <Home color="#112132" />,
          },
          {
            label: 'My trips',
            link: '/lounge/bookings',
            icon: <Cart color="#112132" />,
          },
          {
            label: 'AI Travel companion',
            link: '/companion',
            icon: <Chat color="#112132" />,
          },
        ]}
      />
      <Container
        p={16}
        sx={{
          maxWidth: '375px',
          height: '100%',
        }}
      >
        {children}
      </Container>
    </>
  );
}
