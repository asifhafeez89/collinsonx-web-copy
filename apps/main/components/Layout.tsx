import { Container } from '@mantine/core';
interface LayoutProps {
  children: JSX.Element;
}
const Layout = ({ children }: LayoutProps) => {
  return (
    <Container mt={60} sx={{ maxWidth: '375px' }}>
      {children}
    </Container>
  );
};

export default Layout;