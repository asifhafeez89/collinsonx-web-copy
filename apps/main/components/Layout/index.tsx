import { Container } from '@mantine/core';
interface LayoutProps {
  children: JSX.Element;
}
const Layout = ({ children }: LayoutProps) => {
  return <Container mt={60}>{children}</Container>;
};

export default Layout;
