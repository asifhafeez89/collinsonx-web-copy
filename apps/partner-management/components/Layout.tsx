import colors from '@collinsonx/design-system/colour-constants-partner';
import { Box, Flex, MediaQuery } from '@collinsonx/design-system/core';
import Link from 'next/link';

interface LayoutProps {
  subHeader?: JSX.Element;
  hasPadding?: boolean;
  children: JSX.Element;
}

export default function Layout({
  children,
  subHeader,
  hasPadding = true,
}: LayoutProps) {
  return (
    <div
      style={{
        width: '100%',
        height: '100%',
      }}
    >
      <MediaQuery
        query="print"
        styles={{
          display: 'none',
        }}
      >
        <header
          style={{
            backgroundColor: '#EFEFF1',
            borderBottom: '1px solid #D5D5D5',
            width: '100%',
            padding: '2rem',
            textAlign: 'center',
            position: 'relative',
            margin: 0,
          }}
        >
          <Flex justify="center" align="center">
            <Box
              sx={{
                position: 'absolute',
                right: '60px',
              }}
            >
              <Link href="/auth/signout">Logout</Link>
            </Box>
          </Flex>
        </header>
      </MediaQuery>
      {subHeader && (
        <MediaQuery
          query="print"
          styles={{
            display: 'none',
          }}
        >
          {subHeader}
        </MediaQuery>
      )}
      <main
        style={{
          padding: hasPadding ? '0 64px' : 'auto',
          margin: 0,
          height: '100%',
          width: '100%',
          backgroundColor: colors['partner-bg-surface'],
        }}
      >
        {children}
      </main>
    </div>
  );
}
