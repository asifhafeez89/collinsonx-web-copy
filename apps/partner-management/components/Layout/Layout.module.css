.headerInner {
  position: absolute;
  right: 60px;
}
