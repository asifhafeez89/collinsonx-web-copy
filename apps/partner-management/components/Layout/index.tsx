import colors from '@collinsonx/design-system/colour-constants-partner';
import HeaderNav, { HeaderNavProps } from '@components/HeaderNav';

interface LayoutProps {
  subHeader?: JSX.Element;
  hasPadding?: boolean;
  children: JSX.Element;
  headerNavProps: HeaderNavProps;
}

export default function Layout({
  children,
  headerNavProps,
  hasPadding = true,
}: LayoutProps) {
  return (
    <div
      style={{
        width: '100%',
        height: '100%',
      }}
    >
      <HeaderNav {...headerNavProps} />
      <main
        style={{
          padding: hasPadding ? '24px 64px' : 'auto',
          margin: 0,
          height: 'auto',
          minHeight: 'calc(100% - 65px)',
          width: '100%',
          backgroundColor: colors['bg-surface'],
        }}
      >
        {children}
      </main>
    </div>
  );
}
