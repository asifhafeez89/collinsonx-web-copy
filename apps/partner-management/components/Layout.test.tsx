import Layout from './Layout';
import '@testing-library/jest-dom/extend-expect';
import {
  fireEvent,
  render,
  screen,
  waitFor,
  getNodeText,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';


describe('<Layout />', () => {
    it('should render', () => {
  
      const component = render(
        <Layout    
        subHeader =  {<>header</>}
  hasPadding= {true}
  children = {<></>}
        />
      );
  
      expect(component).toMatchSnapshot();
    });
}
)