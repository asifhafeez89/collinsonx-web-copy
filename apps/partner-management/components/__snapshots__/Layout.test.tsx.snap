// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": @media print {
  .emotion-0 {
    display: none;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  position: absolute;
  right: 60px;
}

<body>
    <div>
      <div
        style="width: 100%; height: 100%;"
      >
        <header
          class="emotion-0 emotion-1"
          style="background-color: rgb(239, 239, 241); border-bottom: 1px solid #D5D5D5; width: 100%; padding: 2rem; text-align: center; position: relative; margin: 0px;"
        >
          <div
            class="emotion-2"
          >
            <div
              class="emotion-3"
            >
              <a
                href="/auth/signout"
              >
                Logout
              </a>
            </div>
          </div>
        </header>
        header
        <main
          style="padding: 24px 64px; margin: 0px; height: auto; min-height: calc(100% - 65px); width: 100%; background-color: rgb(251, 251, 251);"
        />
      </div>
    </div>
  </body>,
  "container": @media print {
  .emotion-0 {
    display: none;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-3 {
  position: absolute;
  right: 60px;
}

<div>
    <div
      style="width: 100%; height: 100%;"
    >
      <header
        class="emotion-0 emotion-1"
        style="background-color: rgb(239, 239, 241); border-bottom: 1px solid #D5D5D5; width: 100%; padding: 2rem; text-align: center; position: relative; margin: 0px;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          >
            <a
              href="/auth/signout"
            >
              Logout
            </a>
          </div>
        </div>
      </header>
      header
      <main
        style="padding: 24px 64px; margin: 0px; height: auto; min-height: calc(100% - 65px); width: 100%; background-color: rgb(251, 251, 251);"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
